pkg/apis/controller/suggestions/v1alpha3/suggestion_types.go

```diff
@@ -25,15 +25,16 @@ import (
 
 // SuggestionSpec defines the desired state of Suggestion
 type SuggestionSpec struct {
+	AlgorithmName string `json:"algorithmName"`
 	// Number of suggestions requested
 	Requests int32 `json:"requests,omitempty"`
-
-	//Algorithm settings set by the user in the experiment config
-	AlgorithmSpec *common.AlgorithmSpec `json:"algorithmSpec,omitempty"`
 }
 
 // SuggestionStatus defines the observed state of Suggestion
 type SuggestionStatus struct {
+	//Algorithm settings set by the user in the experiment config
+	Algorithm *common.AlgorithmSpec `json:"algorithm,omitempty"`
```

Comment:
Do we just need AlgorithmSettings here?

Reply:
Yeah, Actually, we do. I will update it.